code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -116,15 +116,16 @@
 			return
 
 /obj/item/device/lightreplacer/attack_self(mob/user)
-	/* // This would probably be a bit OP. If you want it though, uncomment the code.
-	if(isrobot(user))
-		var/mob/living/silicon/robot/R = user
-		if(R.emagged)
-			src.Emag()
-			to_chat(usr, "You shortcircuit the [src].")
-			return
-	*/
-	to_chat(usr, "It has [uses] lights remaining.")
+	if(!isliving(user) || !do_after(user, 1 SECONDS, src))
+		return
+
+	for(var/obj/machinery/light/O in range(1,user))
+		if (!CanUse(user))
```

Comment:
This should just be an early return outside of the loop instead of a break statement inside

Reply:
Changed it.

Reply:
When I say early return I mean having the return statement **outside of the for loop**

Reply:
Rewrote it a bit to move the check to outside the loop and give a small delay when replacing each light. Does it look better now?